File: cash_drawer/cash_drawer/report/delegate_drawer_closing_summery/delegate_drawer_closing_summery.js
// Copyright (c) 2016, Techstation and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Delegate Drawer Closing Summery"] = {
	"filters": [
		{
			"fieldname": "from_date",
	   		"label": __("From Date"),
			"fieldtype": "Date",
			"default": frappe.datetime.month_start()
		},
		{
			"fieldname": "to_date",
			"label": __("To Date"),
			"fieldtype": "Date",
			"default": frappe.datetime.month_end()
		},
		{
			"fieldname": "delegate",
			"label": __("Delegate"),
			"fieldtype": "Link",
			"options": "Delegate",
		}
	]
};

File: cash_drawer/cash_drawer/report/delegate_drawer_closing_summery/delegate_drawer_closing_summery.py
# Copyright (c) 2013, Techstation and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe import msgprint, _


def execute(filters=None):
    conditions, filters = get_conditions(filters)
    columns = get_column()
    data = get_data(conditions, filters)
    return columns, data


def get_column():
    return [
        _("Date") + ":Date:120",
        _("Delegate") + ":Link/Delegate:150",
        _("ID") + ":Link/Delegate Drawer:150",
        _("Cash On Delivery") + ":Currency:150",
        _("Bank Instalment") + ":Currency:150",
    ]


def get_data(conditions, filters):
    orders = frappe.db.sql(
        """select date(creation), delegate, name, cash_on_delivery, bank_instalment
		 from `tabDelegate Drawer` where 
		docstatus = 1 %s order by creation asc;"""
        % conditions,
        filters,
        as_list=1,
    )
    return orders


def get_conditions(filters):
    conditions = ""
    if filters.get("from_date"):
        conditions += " and creation >= %(from_date)s"
    if filters.get("to_date"):
        conditions += " and creation <= %(to_date)s"
    if filters.get("delegate"):
        conditions += "and delegate = %(delegate)s"

    return conditions, filters
